charabia/src/token.rs

```diff
@@ -161,3 +177,34 @@ impl Token<'_> {
         }
     }
 }
+
+#[cfg(test)]
+// WORKAROUND: The quickcheck macro can't be used with a type with lifetime.
+pub type StaticToken = Token<'static>;
+
+#[cfg(test)]
+impl Arbitrary for Token<'static> {
+    fn arbitrary(g: &mut Gen) -> Self {
+        let lemma = String::arbitrary(g);
+
+        let bytes_count = lemma.len();
+        let byte_start = usize::arbitrary(g).saturating_sub(bytes_count);
+        let byte_end = byte_start + bytes_count;
+
+        let chars_count = lemma.chars().count();
+        let char_start = usize::arbitrary(g).saturating_sub(chars_count);
+        let char_end = char_start + chars_count;
+
+        Token {
+            kind: TokenKind::arbitrary(g),
+            lemma: Cow::Owned(String::arbitrary(g)),
+            char_start,
+            char_end,
+            byte_start,
+            byte_end,
+            char_map: None,
```

Comment:
Using a random incorrect `char_map` here can trigger a lot of out of UTF-8 boundary panic, but I assume that's not something we are interested in.
```rs
            char_map: Option::arbitrary(g),
```

Reply:
Hello @choznerol,

You could run a [normalization over a str](https://docs.rs/charabia/0.8.3/charabia/normalizer/trait.Normalize.html) activating the char_map creation, then check if the method [Token::original_lengths](https://docs.rs/charabia/0.8.3/charabia/struct.Token.html#method.original_lengths) doesn't panic if it's called for any length.

Reply:
> You could run a [normalization over a str](https://docs.rs/charabia/0.8.3/charabia/normalizer/trait.Normalize.html) activating the char_map creation, then check if the method [Token::original_lengths](https://docs.rs/charabia/0.8.3/charabia/struct.Token.html#method.original_lengths) doesn't panic if it's called for any length.

Thanks! Fixed in 2e5e2c3